src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int TABLE_MAGNIFICATION_FACTOR = 2;
+    private int threshold;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node<K, V> intermediateNode = table[index];
+        if (intermediateNode == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        while (intermediateNode != null) {
+            if (Objects.equals(intermediateNode.key, key)) {
+                intermediateNode.value = value;
+                return;
+            }
+            if (intermediateNode.next == null) {
+                intermediateNode.next = new Node<>(key, value);
+                size++;
+                return;
+            }
+            intermediateNode = intermediateNode.next;
+        }
+        intermediateNode.next = new Node<>(key, value);
```

Comment:
тут можна скоротити код, обгорнувши while у блок else
```suggestion
        if (intermediateNode == null) {
            table[index] = new Node<>(key, value);
        } else {
            while (intermediateNode.next != null || Objects.equals(intermediateNode.key, key)) {
                if (Objects.equals(intermediateNode.key, key)) {
                    intermediateNode.value = value;
                    return;
                }
                intermediateNode = intermediateNode.next;
            }
            intermediateNode.next = new Node<>(key, value);
        }
```

Reply:
сократил